core/domain/suggestion_services.py

```diff
@@ -3936,22 +3937,30 @@ def _generate_translation_contributor_certificate_data(
     words_count = 0
     for model in suggestions:
         suggestion = get_suggestion_from_model(model)
+        suggestion_change = suggestion.change
+        if (
+                suggestion_change.cmd == 'add_written_translation' and (
+                    translation_domain.TranslatableContentFormat
+                    .is_data_format_list(suggestion_change.data_format)
+                )
+        ):
+            words_count += len(suggestion_change.translation_html)
+        else:
+            # Retrieve the html content that is emphasized on the
+            # Contributor Dashboard pages. This content is what stands
+            # out for each suggestion when a user views a list of
+            # suggestions.
+            get_html_representing_suggestion = (
+                SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS[
+                    suggestion.suggestion_type]
+            )
+            plain_text = _get_plain_text_from_html_content_string(
+                get_html_representing_suggestion(suggestion))
 
-        # Retrieve the html content that is emphasized on the
-        # Contributor Dashboard pages. This content is what stands
-        # out for each suggestion when a user views a list of
-        # suggestions.
-        get_html_representing_suggestion = (
-            SUGGESTION_EMPHASIZED_TEXT_GETTER_FUNCTIONS[
-                suggestion.suggestion_type]
-        )
-        plain_text = _get_plain_text_from_html_content_string(
-            get_html_representing_suggestion(suggestion))
-
-        words = plain_text.split(' ')
-        words_without_empty_strings = [
-            word for word in words if word != '']
-        words_count += len(words_without_empty_strings)
+            words = plain_text.split(' ')
+            words_without_empty_strings = [
+                word for word in words if word != '']
```

Comment:
the only word in words that can be empty string are the first element in the list and the last one. Therefore, you can check just for these two elements instead of looping trohought all the element of the list.

Reply:
That's not true. Adjacent words in plain_text can have multiple spaces between them, and the split() would then yield multiple empty-string elements. Example:

```
>>> plain_text = 'df ds ds sd     sd'
>>> plain_text.split(' ')
['df', 'ds', 'ds', 'sd', '', '', '', '', 'sd']
```

Reply:
Yeah you are right. I was not thinking about this case.

I was just wondering if we can make this part of code faster but probably not without loosing in simplicity...